src/java.base/share/classes/jdk/internal/foreign/AbstractMemorySegmentImpl.java

```diff
@@ -358,10 +359,19 @@ private <Z> Z toArray(Class<Z> arrayClass, ValueLayout elemLayout, IntFunction<Z
         return arr;
     }
 
+    public enum AccessConstraint {
+        READ_ONLY, ON_WRITE_UOE, ON_WRITE_IAE
```

Comment:
I think the names are a bit confusing. Perhaps having NONE instead of READ_ONLY would be better.